### PR TITLE
feat: Canvas Selected State

### LINKED ISSUE #28420: [Feature]: Sidebar v 1.1
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

Improving the default and selected states of the canvas to improve interactions, and reduce navigation.

Changes to be made:
- [ ] Change the canvas selected state to have the following canvas properties in the property pane
image (3)
- [ ] Change the entity explorer state to change to the explorer menu rather than stick to the widgets menu when a widget is selected

### Why should this be worked on?

This is to help address some concerns about the discoverability of new entities and further ensure that we follow the following clear mental model: properties of selected entities are always on the right side of the canvas.

### PR DESCRIPTION
**Pull request by hetunandu**
## Description

Shows the canvas in the selected state when no widgets are selected. In a future version, when canvas is in the selected state, the left pane will show list of widgets you can drag in.

#### PR fixes following issue(s)
Fixes #28420

#### Media

https://github.com/appsmithorg/appsmith/assets/12022471/444be667-1f5a-459b-b521-590e1552555b

#### Type of change

- New feature (non-breaking change which adds functionality)

- [ ] Manual
- [ ] JUnit
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

## Summary by CodeRabbit

- **New Features**
  - Introduced new visual components for selected states and widget names in the layout system.

- **Refactor**
  - Updated element locator for improved UI interaction in testing framework.

- **Style**
  - Enhanced visual representation with styled components for a better user experience.

### CODE REVIEW

File: app/client/src/layoutSystems/fixedlayout/canvas/MainCanvasSelectedLayer.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import { Layers } from "constants/Layers";
+import { useFeatureFlag } from "utils/hooks/useFeatureFlag";
+import { FEATURE_FLAG } from "@appsmith/entities/FeatureFlag";
+import { previewModeSelector } from "selectors/editorSelectors";
+
+const SelectedState = styled.div<{
+  isSelected: boolean;
+}>`
+  border: 1px solid
+    ${({ isSelected }) => (isSelected ? "#F86A2B" : "transparent")};
```
Comment (albinAppsmith) [inline app/client/src/layoutSystems/fixedlayout/canvas/MainCanvasSelectedLayer.tsx:14]: You can use --ads-v2-color-orange-500 for #F86A2B

**Comment by hetunandu** [comment]
closing in favour of https://github.com/appsmithorg/appsmith/pull/29528

**Pull request closed without merging**